IdentityCore/src/util/ASAuthorizationSingleSignOnProvider+MSIDExtensions.m

```diff
@@ -47,6 +48,7 @@ - (ASAuthorizationSingleSignOnRequest *)createSSORequestWithOperationRequest:(MS
     
     ASAuthorizationSingleSignOnRequest *ssoRequest = [self createRequest];
     ssoRequest.requestedOperation = [operationRequest.class operation];
+    [self.class setRequiresUI:requiresUI forRequest:ssoRequest];
```

Comment:
The default value of userInterfaceEnabled is YES. I'm not sure what the behavior would be if it is left enabled for a silent request. 
Should we add logic to default it according to ssoRequest.operation? Like if the operation is ASAuthorizationOperationLogin default it to YES, NO otherwise.

Reply:
>The default value of userInterfaceEnabled is YES. I'm not sure what the behavior would be if it is left enabled for a silent request.
It will be same behavior as today - UI will flicker when silent request is executed and request will fail when screen is locked.
When this is set to NO, those issues shouldn't happen. 

>Should we add logic to default it according to ssoRequest.operation? Like if the operation is ASAuthorizationOperationLogin default it to YES, NO otherwise.

Can you give a bit more context on this? Thanks!

Reply:
I was suggesting to set ssoRequest.userInterfaceEnabled here based on the type of operationRequest so that we won't need to add requiresUI as an input parameter to createSSORequestWithOperationRequest:

Setting ssoRequest.userInterfaceEnabled to YES if ssoRequest.requestedOperation == [ASAuthorizationOperationLogin](https://developer.apple.com/documentation/authenticationservices/asauthorizationopenidoperation?language=objc) but NO for other operations types. 

If we need to set userInterfaceEnabled to YES for a silent operation we can set it explicitly after calling createSSORequestWithOperationRequest:

Reply:
But in that case, this method would need to know about all possible requested operation types that require UI, no?

Reply:
I was thinking that it would need to know just ASAuthorizationOperationLogin, ie. if operation is not ASAuthorizationOperationLogin then set it to NO.
It seems a bit weird to me that it is YES by default, since it will be NO for majority of operations, right? What would be the downside of keeping it YES for all operations?

Reply:
We also have signout operation that will likely require UX in the future, and device registration operation, so I prefer to not hardcode this just to login and later grow into a list of if-checks :)

>It seems a bit weird to me that it is YES by default, since it will be NO for majority of operations, right? What would be the downside of keeping it YES for all operations?

Yes, Apple wanted to keep this backward compatible per my discussion. Previous behavior is UI is always possible, hence YES is default. 
Downside of keeping YES for all operations is today's behavior - UX flickering and performance issues for apps like Teams and Outlook, since when you keep it as YES, SSO extension invocation will send app to the background even if SSO extension is not handling the request. This creates a weird UX transitioning effect.

Reply:
I see, my concern was mainly if we forget to set it to NO for a request that doesn't require UI hence resolving that by tying its value to the operation. That seems mostly improbable, resolving this conversation :)